runtime/shared_common/OSCache.cpp

```diff
@@ -1285,9 +1295,12 @@ SH_OSCache::generateCacheUniqueID(J9VMThread* currentThread, const char* cacheDi
 	getCacheVersionAndGen(PORTLIB, vm, nameWithVGen, J9SH_MAXPATH, cacheName, &versionData, OSCACHE_CURRENT_CACHE_GEN, true, layer);
 	/* Directory is included here, so if the cache directory is renamed, caches with layer > 0 becomes unusable */
 	getCachePathName(PORTLIB, cacheDirName, cacheFilePathName, J9SH_MAXPATH, nameWithVGen);
-	I_64 timeStamp = j9file_lastmod(cacheFilePathName);
 	I_64 fileSize = j9file_length(cacheFilePathName);
-	return j9str_printf(PORTLIB, buf, bufLen, "%s-%llx_%llx", cacheFilePathName, fileSize, timeStamp);
+	if (NULL != buf) {
+		UDATA bufLenrequired = j9str_printf(PORTLIB, NULL, 0, "%s-%llx_%llx_%zx_%zx_%zx_%zx", cacheFilePathName, fileSize, createtime, metadataBytes, classesBytes, lineNumTabBytes, varTabBytes);
```

Comment:
As already mentioned for the tracepoints, %llx is not correct for UDATA, use %zx. %llx is correct for U_64

Reply:
Sorry, ignore this comment, I see it is correct.